source/databricks/package/balance_fixing_total_production.py

```diff
@@ -184,9 +188,115 @@ def _get_grid_areas_df(cached_integration_events_df, batch_grid_areas_df) -> Dat
     return grid_area_df
 
 
+def _get_master_basis_data_df(
+    metering_points_periods_df: DataFrame,
+    market_roles_periods_df: DataFrame,
+    grid_area_df: DataFrame,
+    period_start_datetime,
+    period_end_datetime,
+) -> DataFrame:
+    metering_points_in_grid_area = metering_points_periods_df.join(
+        grid_area_df,
+        metering_points_periods_df["GridArea"] == grid_area_df["GridAreaCode"],
+        "inner",
+    )
+
+    metering_point_periods_df = (
+        metering_points_in_grid_area.where(col("FromDate") <= period_end_datetime)
```

Comment:
Shouldn't it be like this?
```suggestion
        metering_points_in_grid_area.where(col("FromDate") < period_end_datetime)
```

Reply:
no. As period_end_datetime is not inclusive in the period.